docs/architecture/component_templates.md

```diff
@@ -0,0 +1,141 @@
+# Account component templates
+
+An account component template provides a general description of an account component. It encapsulates all the information needed to initialize and manage the component.
+
+Specifically, a template specifies a component's **metadata** and its **code**.
+
+Once defined, a component template can be instantiated as account components, which can then be merged to form the account's `Code`.
+
+## Component code
+
+The component template’s code defines a library of functions that operate on the specified storage layout.
+
+## Component metadata
+
+The component metadata describes the account component entirely: its name, description, version, and storage layout.
+
+The storage layout must specify a contiguous list of slot values that starts at index `0`, and can optionally specify initial values for each of the slots. Alternatively, placeholders can be utilized to identify values that should be provided at the moment of instantiation.
+
+### TOML specification
+
+The component metadata can be defined using TOML. Below is an example specification:
+
+```toml
```

Comment:
nit: if possible, I'd try to format the example so that it doesn't result in horizontal scroll bars.